Lib/sqlite3/test/dbapi.py

```diff
@@ -724,6 +728,21 @@ def run(cur, errors):
         if len(errors) > 0:
             self.fail("\n".join(errors))
 
+    def test_dont_check_same_thread(self):
```

Comment:
Please, decorate this with `@threading_helper.reap_threads`

Reply:
I was unaware of that helper; thanks! I guess we should apply that to all the test methods in the `ThreadTests` class.

Reply:
(Digression: All of `ThreadTests` should be refactored, IMO.)

Reply:
> (Digression: All of `ThreadTests` should be refactored, IMO.)

Go ahead, but please explain very well why are you refactoring them as refactors almost always implies risk of regressions and errors.

Reply:
Of course. I'll do it in a separate issue/PR, though.

Reply:
Helper added in af0fc7ef7af4e12f9660e6201af1e8d584fd2629. Marking this as resolved.

Reply:
> > (Digression: All of `ThreadTests` should be refactored, IMO.)
> 
> Go ahead, but please explain very well why are you refactoring them as refactors almost always implies risk of regressions and errors.

PR created at #26748